docs/source/developers/guide/step_by_step/pr_and_github.rst

```diff
@@ -24,14 +24,171 @@
+   the name of the branch that you pushed and next to it a green button
+   **Compare & Pull Request**. After clicking on it, you should add a
+   title and description of the Pull Request. Underneath you can check
+   once again the changes you have made.
+
+   .. seealso::
+      
+      Get more details on naming the Pull Request in Arrow repository
+      and other additional information :ref:`pull_request_and_review`
+      section.
+
+Reviews and merge of the Pull Request
+=====================================
+
+When the Pull Request is submitted it waits to get reviewed. One of
+great things about Open Source is your work gets lots of feedback and
+so it gets perfected. Do not be discouraged by the time it takes for
+the PR to get merged due to reviews and corrections. It is a process
+that supports quality and with it you can learn a lot.
+
+If it still takes too long to get merged, do not hesitate to remind
+maintainers in the comment section of the Pull Request and post
+reminders on the JIRA ticket also.
+
+How to get your Pull Request to be reviewed?
+--------------------------------------------
+
+Arrow maintainers will be notified when a Pull Request is created and
+they will get to it as soon as possible. If days pass and it still had
+not been reviewed go ahead and mention the reporter of the JIRA issue 
+or a developer that you communicated with via JIRA comments, mailing
+list or GitHub.
+
+To put a **mention** in GitHub insert @ in the comment and select the
+username from the list.
+
+Commenting on a Pull Request
+----------------------------
+
+When a Pull Request is open in the repository you and other developers
+can comment on the proposed solution.
+
+To create a general comment navigate to the **Conversation** tab of
+you Pull Request and start writing in the comment box at the bottom of
+the page.
+
+You can also comment on a section of the file to point out something
+specific from your code. To do this navigate to **Files changed** tab and
+select a line you want to insert the comment to. Hovering over the beginning
+of the line you will see a **blue plus icon**. You can click on it or drag
+it to select multiple lines and then click the icon to insert the comment.
+
+Resolve conversation
+--------------------
+
+You can resolve a conversion in a Pull Request review by clicking
+**Resolve conversation** in the **Files changed** tab. This way the
+conversation will be collapsed and marked as resolved which will make it
+easier for you to organize what is done and what still needs to be addressed.
+
+After getting a review
+----------------------
+
+The procedure after getting reviews is similar to creating the initial Pull Request.
+You need to update your code locally, make a commit, update the branch to sync
+it with upstream (or origin if there were commits from other developers or if you 
+committed suggestions from the GitHub) and push your code to origin. It will
```

Comment:
I am wondering if we want to spell out the commands for this?

Reply:
I would agree, it would be nice to have everything in one place and remove duplicated info from the contributing page.

Reply:
The contributing page lists how to rebase your PR, but not how to fetch commits from your remote fork in case you accepted suggestions. That's maybe something that could be added here, and how to consolidate it with contributing.rst can be something to do later?

Reply:
Yes, that's right. The pull from origin is only suggested in the contributing.rst. I will make it explicit and add it here!